components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/logout/idpinit/processor/FederatedIdpInitLogoutProcessor.java

```diff
@@ -0,0 +1,632 @@
+                return new LogoutResponse.LogoutResponseBuilder(HttpServletResponse.SC_INTERNAL_SERVER_ERROR,
+                        ErrorMessages.LOGOUT_SERVER_EXCEPTION.getMessage());
+            }
+            UserSessionManagementService userSessionManagementService =
+                    OpenIDConnectAuthenticatorDataHolder.getInstance()
+                            .getUserSessionManagementService();
+            userSessionManagementService.terminateSessionsByUserId(userId);
+            if (log.isDebugEnabled()) {
+                log.debug("Sessions terminated for user Id: " + userId);
+            }
+            return new LogoutResponse.LogoutResponseBuilder(HttpServletResponse.SC_OK,
+                    OIDCAuthenticatorConstants.BackchannelLogout.LOGOUT_SUCCESS);
+        } catch (SessionManagementException e) {
+            throw handleLogoutServerException(ErrorMessages.USER_SESSION_TERMINATION_FAILURE, e, userId);
+        }
+    }
+
+    /**
+     * Validate the JWT token signature and the mandatory claim according to the OIDC specification.
+     *
+     * @param signedJWT
+     * @param identityProvider
+     * @return boolean value
+     * @throws LogoutClientException
+     * @throws LogoutServerException
+     */
+    private void validateLogoutToken(SignedJWT signedJWT, IdentityProvider identityProvider)
+            throws LogoutClientException, LogoutServerException {
+
+        try {
+            JWTClaimsSet claimsSet = signedJWT.getJWTClaimsSet();
+
+            if (!JWTSignatureValidationUtils.validateSignature(signedJWT, identityProvider)) {
+                throw new LogoutClientException(ErrorMessages.LOGOUT_TOKEN_SIGNATURE_VALIDATION_FAILED.getCode(),
+                        ErrorMessages.LOGOUT_TOKEN_SIGNATURE_VALIDATION_FAILED.getMessage());
+            }
+            validateAudience(claimsSet.getAudience(), identityProvider);
+            validateIat(claimsSet.getIssueTime());
+            validateEventClaim((JSONObject) claimsSet.getClaim(OIDCAuthenticatorConstants.Claim.EVENTS));
+            validateNonce(claimsSet);
+        } catch (ParseException e) {
+            throw handleLogoutClientException(ErrorMessages.LOGOUT_TOKEN_PARSING_FAILURE, e);
+        } catch (JOSEException | IdentityOAuth2Exception e) {
+            throw handleLogoutServerException(ErrorMessages.LOGOUT_TOKEN_SIGNATURE_VALIDATION_FAILED, e);
+        }
+    }
+
+    /**
+     * Retrieve userId of the federated user.
+     *
+     * @param tenantDomain     - tenant domain of the logout request.
+     * @param sub              - sub claim in the logout token.
+     * @param identityProvider - identity provider.
+     * @return
+     * @throws LogoutServerException
+     */
+    private String getUserId(String tenantDomain, String sub, IdentityProvider identityProvider)
+            throws LogoutServerException {
+
+        try {
+            int tenantId = tenantDomain == null ? -1 : IdentityTenantUtil.getTenantId(tenantDomain);
```

Comment:
IdentityTenantUtil.getTenantId(tenantDomain) 

doesn't this method handle null and invalid tenants?